rasa/core/brokers/broker.py

```diff
@@ -3,13 +3,20 @@
 from asyncio import AbstractEventLoop
 from typing import Any, Dict, Text, Optional, Union
 
+import aio_pika.exceptions
+import psycopg2
+
 import rasa.shared.utils.common
 import rasa.shared.utils.io
+from rasa.shared.exceptions import ConnectionException
 from rasa.utils.endpoints import EndpointConfig
 
 logger = logging.getLogger(__name__)
 
 
+CONNECTION_ERRORS = (psycopg2.OperationalError, aio_pika.exceptions.AMQPConnectionError)
```

Comment:
questions for this constant and the other one below:
- do we care about importing modules like `psycopg2`, `pymongo`, etc... at the top-level? Is it impacting the start-up time of the CLI and do we have evidence of that?
- would it make sense to have only one shared `CONNECTION_ERRORS` tuple somewhere in `rasa.shared.constants` or `rasa.constants`? I'm sitting between that and defining the constant closer to the the code that needs it.

Reply:
@wochinge I dunno if you saw this. Any opinion?

Reply:
Very good point. I think **in theory** the `broker` / `tracker_store` modules should only be imported when they are actually used. Which means that they shouldn't impact things like `rasa --help` but of course will impact things like `rasa run` (which is fine). From that perspective it's in my opinion fine to have the imports of the exceptions as top level imports. 

You could argue that **in practise** we have no clue when / where these modules are imported and that it could be easier to use local imports in this one module instead of having to watch out in all other modules. 

As far as I know these 3rd party imports are only used in a few modules so I think that the 2nd approach (using local imports) might indeed be the easier one. 

>  Is it impacting the start-up time of the CLI and do we have evidence of that?

we once had a test for the CLI startup time but removed it as far as I know 😬 

> would it make sense to have only one shared CONNECTION_ERRORS tuple somewhere in rasa.shared.constants or rasa.constants?

I personally would favor locality. Downstream dependencies (like Rasa X) should handle their connection issues themselves (which speaks against `shared`) and I think it makes sense to keep the 3rd party integrations as much as possible separate which would speak against the `rasa.constants` (the last point is a rather weak argument to be honest).

Reply:
I will defer to your advice here. I am not a big fan of having local imports here and there without a clear understanding of why we're doing this (right it's more or less a guess). That'd be way better if we could make educated decisions using performance arguments and data points. Changes made in 230c773d4b0